go/vt/vttablet/tabletserver/tx_pool_test.go

```diff
@@ -265,6 +265,25 @@ func TestTxPoolTransactionIsolation(t *testing.T) {
 	}
 }
 
+func TestTxPoolAutocommit(t *testing.T) {
+	db := fakesqldb.New(t)
+	defer db.Close()
+	txPool := newTxPool()
+	txPool.Open(db.ConnParams(), db.ConnParams(), db.ConnParams())
+	ctx := context.Background()
+
+	// Start a transaction with autocommit. This will ensure that the executor does not send begin/commit statements
+	// to mysql.
+	txid, err := txPool.Begin(ctx, &querypb.ExecuteOptions{TransactionIsolation: querypb.ExecuteOptions_AUTOCOMMIT})
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = txPool.Commit(ctx, txid, &fakeMessageCommitter{})
```

Comment:
Other than testing that this actually "works", there's nothing that indicates whether or not it does something different from the normal case.

IMO we should add some validation that no queries actually went to mysql OR drop this test since it doesn't actually do any real testing.

Reply:
I think the test is actually doing what we expect. I added a comment explaining why. Let me know if that makes sense.